library/ha_auth.c

```diff
@@ -78,16 +72,21 @@ ziti_auth_method_t *new_ha_auth(uv_loop_t *l, const char* url, tls_context *tls)
     return &auth->api;
 }
 
-static int ha_set_endpoint(ziti_auth_method_t *self, const char *url) {
-    struct ha_auth_s *auth = HA_AUTH(self);
-    char *ep = strdup(url);
+static char *internal_oidc_path(const char *url) {
     struct tlsuv_url_s u;
-    tlsuv_parse_url(&u, ep);
-    if (u.path) {
-        *(char*)u.path = '\0';
-    }
+    tlsuv_parse_url(&u, url);
+
+    size_t baselen = u.path ? u.path - url : strlen(url);
```

Comment:
isn't this just u.hostname + u.hostname_len

Reply:
i guess i'm relaying a worry about what happens if the implementation ever changes, trying to keep this entirely to the parsed struct

Reply:
no